### ISSUE TITLE
FAIL: EFI Firmware while running "compatibility check script" on 4th Gen Intel CPU

### ISSUE THREAD

**Issue by slattann** [clearlinux/distribution#364]
**Describe the bug**
Clear Linux Compatibility test errors out "EFI Firmware not compatible"

 on Intel® Core™ i7-4790K CPU @4.0GHz.

**To Reproduce**
1. Steps followed to create a USB boot disk: https://clearlinux.org/documentation/clear-linux/get-started/bare-metal-install
2. On Step 7 when Clear Linux selected throws an errors:
Error loading /EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2-683: Not Found
Failed to execute Clear Linux OS (/EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2.683): Not Found
3. Then tried checking the compatibility  for Clear Linux here are results:

**Expected behavior**
Should be able to install like any other Linux distro. Eg: I could able to install Ubuntu 18.04 on the same machine.

**Screenshots**
![efi-error](https://user-images.githubusercontent.com/25810508/51311902-04298500-1a70-11e9-9f4f-6f1cc6a89ec8.JPG)

**Environment (please complete the following information):**
 - Clear Linux OS version: clear-27300-installer.img

**Comment by gvancuts** [comment]
I believe this rather indicates that your platform uses a traditional BIOS and not an EFI firmware. Some platforms support both so please make sure you try to boot it in EFI mode (and not legacy Bios).

**Comment by miguelinux** [comment]
Please use the compatibility check script, you can find it at

https://clearlinux.org/documentation/clear-linux/get-started/compatibility-check

to verify is you are using the minimum requirement, including EFI
firmware.

**Comment by miguelinux** [comment]
Oh, just see the screnshot.

@slattann 

Please enable EFI firmware in your system, Clear Linux boots using EFI firmware.

**Comment by ahkok** [comment]
The errors you posted here:
```
Error loading /EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2-683: Not Found
Failed to execute Clear Linux OS (/EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2.683): Not Found
```
Indicate an issue with the image we're publishing.

The screenshot you posted indicates that you boot Ubuntu in Legacy BIOS mode, while Clear Linux OS expects you to boot in UEFI mode. I assume that you are able to switch to UEFI mode, other wise you would not have seen the error messages above.

So, the image appears to be broken. 

However, when I download the image you linked above, and extract it, and boot it in Qemu, I can verify that the error path above *exists* and the image *boots* properly in UEFI mode.

```
root@clr-f3e9a565468f4a599e77ddad3daeccca /boot # cat loader/entries/Clear-linux-native-4.20.2-683.conf
title Clear Linux OS
linux /EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2-683
options root=PARTUUID=78736666-b0b9-405e-9c03-dae79f92f73e quiet console=tty0 console=ttyS0,115200n8 init=/usr/bin/initra-desktop  initcall_debug tsc=reliable no_timer_check noreplace-smp kvm-intel.nested=1 rootfstype=ext4,btrfs,xfs intel_iommu=igfx_off cryptomgr.notests rcupdate.rcu_expedited=1 rcu_nocbs=0-64 rw nomodeset i915.modeset=0
root@clr-f3e9a565468f4a599e77ddad3daeccca /boot # ls -ld /boot/EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2-683
-rwx------ 1 root root 7922848 Jan 16 20:21 /boot/EFI/org.clearlinux/kernel-org.clearlinux.native.4.20.2-683
```

In other words, there appears to be nothing wrong with the image. I can't reproduce the error, nor do I have issues booting the image.

**Comment by AHewitt** [comment]
I have this issue also, I'm on a Macbook Pro running Virtualbox.

**Comment by miguelinux** [comment]
enable **EFI** in VirtualBox 

take a look at:

https://clearlinux.org/documentation/clear-linux/get-started/virtual-machine-install/virtualbox-cl-installer

**Comment by AHewitt** [comment]
That worked! Thank you miguelinux

**Comment by gbodley** [comment]
I tried to install this morning (190521) from USB using clear-29520-live-desktop.iso and got the message of EFI software missing.   Personally I see it as an installer issue with Clear.  I'd rather not start fooling with the BIOS as every other Linux distribution seems to install more or less OK, but GRUB has gotten messed up many times.

**Comment by mdhorn** [comment]
@gbodley 
The Live image will boot in either UEFI or Legacy mode to let you try out Clear Linux* without installing. This is only possible as we've created a Hybrid ISO image to support this.

Clear Linux OS does have a hard requirement of using UEFI to boot.
This is why the installer has a precheck for the required hardware features as well as checking if UEFI is enabled. If we didn't have this check the install would succeed, but you would not be able to boot the newly installed image.

**Comment by gbodley** [comment]
I checked after I posted and I am booting in Kubuntu in UEFI from Partition 1 which is formatted as an EFI boot partition.  All I know is Clear is the only distribution I've tried that has this issue.  I'm no computer expert particularly when it comes to GRUB issues.  But thanks for your quick response.  I'm not completely positive, but I believe the BIOS is set to boot in UEFI mode.  Is there a Linux terminal command I can use to determine the boot method it is using presently.
Here's this:
jl@jl-2n1:~$ ls /sys/firmware/efi
config_table  efivars  esrt  fw_platform_size  fw_vendor  runtime  runtime-map  systab  vars

if it means anything to you.

**Comment by gbodley** [comment]
> @gbodley
> The Live image will boot in either UEFI or Legacy mode to let you try out Clear Linux* without installing. This is only possible as we've created a Hybrid ISO image to support this.
> 
> Clear Linux OS does have a hard requirement of using UEFI to boot.
> This is why the installer has a precheck for the required hardware features as well as checking if UEFI is enabled. If we didn't have this check the install would succeed, but you would not be able to boot the newly installed image.

I did reboot and checked the BIOS settings and as near as I can tell it's set to boot UEFI without secure boot.

**Comment by KTCyankees** [comment]
I am having the same issue as screenshot shows above. (see below).I have tried on multiple systems (HP/Dell i7 PC/Laptops and Enterprise Server Hardware (I work on EMC systems and have tried setting up on XIO controllers, Avamar and RecoverPoint, all controllers/servers that I use to run other Linux version for testing hardware and settings..) the Server version (CLI) works fine so it seems to be related to GUI launching in live (Desktop) version? 
Does not matter if I am using UEFI support or Enhanced, etc... Just cannot get it to work other than in my VM environments running in Windows 10 (VirtualBox) and on a Cisco UCS with ESXi. In both, when I attempt to install the "live" version on the virtual drive(s) it fails with some GUI environmental error (sorry forgot to document it) 
Below is error when running the pre-check config.sh on every PC and server I have tried (6 in total).
I can send more info later if needed but basically it never boots up the live GUI. Just black screen with what looks like a cursor. Every other "live" linux version (Mint, Zorin, CentOS...) work just fine.

XtremIO controller currently running Zorin:
kurt@Zorin2:~/Downloads$ chmod +x clear-linux-check-config.sh
kurt@Zorin2:~/Downloads$ ./clear-linux-check-config.sh host
Checking if host is capable of running Clear Linux* OS

SUCCESS: 64-bit CPU (lm)
SUCCESS: Supplemental Streaming SIMD Extensions 3 (ssse3)
SUCCESS: Streaming SIMD Extensions v4.1 (sse4_1)
SUCCESS: Streaming SIMD Extensions v4.2 (sse4_2)
SUCCESS: Advanced Encryption Standard instruction set (aes)
SUCCESS: Carry-less Multiplication extensions (pclmulqdq)
FAIL: EFI firmware

**Comment by gbodley** [comment]
jl@jl-2n1:~$ ls
clear-linux-check-config.sh  Desktop  Documents  Downloads  dwhelper  Music  Pictures  Public  ScreenRotator  Templates  Videos
jl@jl-2n1:~$ ./clear-linux-check-config.sh
ERROR: Invalid type specified: '' (specify 'host' or 'container')
jl@jl-2n1:~$

**Comment by gbodley** [comment]
jl@jl-2n1:~$ ./clear-linux-check-config.sh host
Checking if host is capable of running Clear Linux* OS

SUCCESS: 64-bit CPU (lm)
SUCCESS: Supplemental Streaming SIMD Extensions 3 (ssse3)
SUCCESS: Streaming SIMD Extensions v4.1 (sse4_1)
SUCCESS: Streaming SIMD Extensions v4.2 (sse4_2)
SUCCESS: Advanced Encryption Standard instruction set (aes)
SUCCESS: Carry-less Multiplication extensions (pclmulqdq)
SUCCESS: EFI firmware
jl@jl-2n1:~$ 

I learned from the post above mine that I was missing the word "host" which of course I would have no way of knowing or understanding.  However the installer running on a live Clear Linux USB apparently reports back false information as it claims there is "EFI firmware missing"

**Comment by KTCyankees** [comment]
Now able to install it to virtual disk in VB. It failed twice on one system (ASUS ROG) that I run most VMs on from EMC but it worked on the UCS and now on another Dell XPS...

Tried creating another USB with live version but same issue on another server (Quanta), Limited on time digging into it the issue so maybe I am missing something. I really just want to use Clear to test Intel based system hardware. Not a big deal. Will wait it out I guess..

**Comment by gbodley** [comment]
I've only got this one Dell to play with and it's my primary computer.

**Comment by KTCyankees** [comment]
I have them all with screens and Ethernet. No wireless on most of these systems but everything else is typical setups when I am installing them for reasons other than the systems they come from. The thing that makes me think I am missing something (maybe even simple) is that I am able to run other "live" version on them. CentOS is the one I use the most but also more consumer focused version like Mint and Zorin. I just cannot work on it now. I did ask a tech to try setting it up and he had the same issue on a VNX control station (RAID and no RAID) but when he runs it on Dell XPS laptop he gets what said was unrelated hardware compatibility error. I am not expert in Linux but I use it every day as a tool or the OE on the systems we test and setup.

**Comment by gbodley** [comment]
Like I said, I have a beautiful Kubuntu install which is perfect in most every respect.  I would like to try Clear because it's fast and looks nice, but it has to be easy to install for me to maintain interest.

**Comment by KTCyankees** [comment]
I should add I am able to install any other versions but the live version is what I would run to boot up systems to run the diags on them. Not looking to install it. Just using live version from USB as a tool. I like it from the system I did install and the VM I setup. Being Intel I am hoping it is a better tool for testing Intel dominated hardware

**Comment by gbodley** [comment]
I have a persistent USB version of Ubuntu Budgie that runs very nicely.  At this point Clear is at best experimental in nature.  You can use MKUSB to create the persistence.  Really Intel needs to make their hardware more compatible with Linux.  That's the actual issue, and has been for quite some time.

**Comment by VadymVolos** [comment]
Hi guys!

> Clear Linux (clear-31040-live-desktop.iso), VirtualBox-6.0.12-133076-Win


I have **a black screen**, and then after five minutes **an error:**

> [ FAIL ] FATAL : Detected missing Required CPU Feature: Streaming SIMD Extension v4.2 (sse4_2)

I checked my processor utility CPU-Z
It support SSE4.1. But does not support SSE4.2.

Maybe you have a black screen for that reason.

----

Check processor and EFI firmware compatibility:
https://docs.01.org/clearlinux/latest/get-started/compatibility-check.html

Discussion of this problem on the forum:
https://community.clearlinux.org/t/installation-in-virtualbox-live-image-issues/552/5
